src/ZFTool/Controller/DiagnosticsController.php

```diff
@@ -224,13 +256,76 @@ public function runAction()
             } else {
                 $model->setErrorLevel(0);
             }
-        } else {
+            return $model;
+    }
+
+    private function processHttpRequest(Request $request, Collection $results)
+    {
+        $defaultAccept = new Accept();
+        $defaultAccept->addMediaType(self::CONTENT_TYPE_HTML);
+
+        $acceptHeader = $request->getHeader('Accept', $defaultAccept);
+
+        $viewModel = function () use ($results) {
```

Comment:
Build the view model directly? The overhead is the same anyway (a closure is an object)